tests/backup_restore_test.php

```diff
@@ -0,0 +1,181 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * BBB Library tests class.
+ *
+ * @package   mod_bigbluebuttonbn
+ * @copyright 2018 - present, Blindside Networks Inc
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @author    Laurent David (laurent@call-learning.fr)
+ */
+
+namespace mod_bigbluebuttonbn;
+
+use calendar_event;
+use context_module;
+use mod_bigbluebuttonbn\logger;
+use mod_bigbluebuttonbn\test\testcase_helper_trait;
+use mod_bigbluebuttonbn_mod_form;
+use MoodleQuickForm;
+use navigation_node;
+use ReflectionClass;
+use restore_date_testcase;
+use stdClass;
+
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+require_once($CFG->libdir . "/phpunit/classes/restore_date_testcase.php");
+
+/**
+ * BBB Library tests class.
+ *
+ * @package   mod_bigbluebuttonbn
+ * @copyright 2018 - present, Blindside Networks Inc
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @author    Laurent David (laurent@call-learning.fr)
+ */
+class backup_restore_test extends restore_date_testcase {
+    use testcase_helper_trait;
+
+    /**
+     * Setup basic
+     */
+    public function setUp(): void {
+        parent::setUp();
+        $this->basic_setup();
+    }
+
+    /**
+     * @dataProvider type_provider
+     */
+    public function test_backup_restore($type) {
```

Comment:
Please define all return types.
This one should be :void

Reply:
Yes, I often leave it out, I need to adjust to the new convention here.